### PR TITLE
Fix posthog identify

### PR DESCRIPTION
**Pull request by timgl**
## Changes

- We were resetting the user every time the distinct id was different, even if that ID was anonymous.
- Also add tracking to login etc pages.

Would be nice to have a is_anonymous function on posthog-js or something but this is good enough for now.

cc @macobo b/c he made this change so checking if there was another reason I'm missing.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: posthog/models/dashboard_item.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 from django.db import models
 from django.db.models.signals import pre_save
 from django.dispatch import receiver
+from sentry_sdk import capture_exception
```
Comment (macobo) [inline posthog/models/dashboard_item.py:5]: Unrelated change?

File: frontend/src/layout.ejs [hunk 2]
```diff
@@ -5,9 +5,15 @@
     <meta name="robots" content="noindex">
     <title>PostHog</title>
     {% include "head.html" %}
+
+    <script>
+        !function(t,e){var o,n,p,r;e.__SV||(window.posthog=e,e._i=[],e.init=function(i,s,a){function g(t,e){var o=e.split(".");2==o.length&&(t=t[o[0]],e=o[1]),t[e]=function(){t.push([e].concat(Array.prototype.slice.call(arguments,0)))}}(p=t.createElement("script")).type="text/javascript",p.async=!0,p.src=s.api_host+"/static/array.js",(r=t.getElementsByTagName("script")[0]).parentNode.insertBefore(p,r);var u=e;for(void 0!==a?u=e[a]=[]:a="posthog",u.people=u.people||[],u.toString=function(t){var e="posthog";return"posthog"!==a&&(e+="."+a),t||(e+=" (stub)"),e},u.people.toString=function(){return u.toString(1)+".people (stub)"},o="capture identify alias people.set people.set_once set_config register register_once unregister opt_out_capturing has_opted_out_capturing opt_in_capturing reset isFeatureEnabled onFeatureFlags".split(" "),n=0;n<o.length;n++)g(u,o[n]);e._i.push([i,s,a])},e.__SV=1)}(document,window.posthog||[]);
+        posthog.init({{js_posthog_api_key | safe}}, {api_host: {{js_posthog_host | safe}}})
```
Comment (macobo) [inline frontend/src/layout.ejs:11]: Will this lead to double initialization?

**Review by macobo — COMMENTED** [review]
LGTM though there's an unintended change + one question.

File: frontend/src/layout.ejs (hunk 2 above)
Comment (timgl) [reply frontend/src/layout.ejs:11]: We only use layout on files that aren't js/webpack etc, so no, but good to check!
image
